scrapy/downloadermiddlewares/redirect.py

```diff
@@ -55,28 +55,25 @@ class RedirectMiddleware(BaseRedirectMiddleware):
 
     def process_response(self, request, response, spider):
         if (request.meta.get('dont_redirect', False) or
-               response.status in getattr(spider, 'handle_httpstatus_list', []) or
-               response.status in request.meta.get('handle_httpstatus_list', []) or
-               request.meta.get('handle_httpstatus_all', False)):
+                response.status in getattr(spider, 'handle_httpstatus_list', []) or
+                response.status in request.meta.get('handle_httpstatus_list', []) or
+                request.meta.get('handle_httpstatus_all', False)):
             return response
 
-        if request.method == 'HEAD':
-            if response.status in [301, 302, 303, 307] and 'Location' in response.headers:
-                redirected_url = urljoin(request.url, response.headers['location'])
+        location = None
+        if 'Location' in response.headers:
+            location = response.headers['location'].decode('latin1')
+
+        if location is not None and response.status in [301, 302, 303, 307]:
+            redirected_url = urljoin(request.url, location)
```

Comment:
Hmmm, right, this is tricky.

So, this is what the server logs say:

```
127.0.0.1 - - [10/Sep/2015 18:38:48] "GET /path_redir_latin1 HTTP/1.1" 302
127.0.0.1 - - [10/Sep/2015 18:38:48] "GET /a%C3%A7%C3%A3o HTTP/1.1" 200
```

and:

```
127.0.0.1 - - [10/Sep/2015 18:39:26] "GET /query_redirect_latin1 HTTP/1.1" 302
127.0.0.1 - - [10/Sep/2015 18:39:26] "GET /something?q=a%C3%A7%C3%A3o HTTP/1.1" 200
```

Which is percent-escaped UTF-8:

```
>>> print urllib.unquote('a%C3%A7%C3%A3o').decode('utf-8')
ação
```

So now I have to see about making sure the behavior will be consistent in Python 2 and 3.